R/multi.point.HR.table.R

```diff
@@ -1,4 +1,9 @@
-multi.point.HR.table <- function(all.groups = NULL, all.survtime = NULL, all.survstat = NULL, truncation.thresholds = c(5, 10), covariates = NULL) {
+multi.point.HR.table <- function(all.groups = NULL, all.survtime = NULL, all.survstat = NULL, truncation.thresholds = NULL, covariates = NULL) {
+	# define default truncation.thresholds as the median survival time and maximum
+	if (is.null(truncation.thresholds)) {
+		fit <- survfit(Surv(all.survtime, all.survstat) ~ 1);
+		truncation.thresholds <- c(summary(fit)$table['median'], max(all.survtime, na.rm = TRUE));
```

Comment:
I suggest using `median(all.survtime, na.rm = TRUE` instead of `summary(fit)$table['median']`, since the latter will be `NA` if < 50% of patients have the event occur.

For example, I think the following example would cause your code to break:

```
library(survival);
library(BoutrosLab.statistics.survival);

data(cancer, package="survival")
fit <- survfit(Surv(time, status) ~ 1, data=rats);
summary(fit);
summary(fit)$table['median']; # NA

truncation.thresholds <- c(summary(fit)$table['median'], max(rats$time, na.rm = TRUE));

# gives error since median survival is NA
multi.point.HR.table(
    all.groups = rats$sex,
    all.survtime = rats$time,
    all.survstat = rats$status,
    truncation.thresholds = truncation.thresholds
    );
```

Reply:
Thanks for the suggestion! It's fixed!